libs/s25main/world/MapBase.cpp

```diff
@@ -87,6 +88,22 @@ MapPoint MapBase::GetNeighbour2(const MapPoint pt, unsigned dir) const
     return MakeMapPoint(::GetNeighbour2(Position(pt), dir));
 }
 
+std::vector<MapPoint> MapBase::GetAllNeighboursUnion(const std::vector<MapPoint>& points) const
+{
+    std::set<MapPoint, MapPointLess> ptsToReturn;
+
+    for(const MapPoint& curMapPt : points)
+    {
+        ptsToReturn.insert(curMapPt);
+        for(const MapPoint neighbourPt : GetNeighbours(curMapPt))
+        {
+            ptsToReturn.insert(neighbourPt);
+        }
```

Comment:
You can (optionally) make it more succinct:
```suggestion
        const auto neighbours = GetNeighbours(curMapPt);
        ptsToReturn.insert(neighbours.begin(), neighbours.end());
```

Or leave the braces:
```suggestion
        for(const MapPoint neighbourPt : GetNeighbours(curMapPt))
            ptsToReturn.insert(neighbourPt);
```